### PR TITLE
Add notarization details to changelog

### PR DESCRIPTION
**Pull request by mdeggies**
@RussellRollins successfully integrated our signing/notarizing process into the CI pipeline, and it will be part of the next release. This PR simply updates the changelog to reflect those changes.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,15 @@
 ## 0.12.18 (Unreleased)
 
+NOTES:
+
+* cli: Our darwin releases for this version and up will be signed and notarized according to Apple's requirements.
+
+    Prior to this release, MacOS 10.15+ users attemping to run our software [reported](https://github.com/hashicorp/terraform/issues/23033) seeing the error: "'terraform' cannot be opened because the developer cannot be verified." This error affected all MacOS 10.15+ users who downloaded our software directly via web browsers, and was caused by [changes to Apple's third-party software requirements](https://developer.apple.com/news/?id=04102019a).
+
+    Users will continue to see an error when attempting to double click on the unzipped binary. This is a bug Apple is working to fix. However, double clicking on the binary is not a recommended way of interacting with our software- users should instead invoke Terraform commands via the command line. [Our recommended approach to install the Terraform CLI can be found here.](https://learn.hashicorp.com/terraform/getting-started/install)
```
Comment (mdeggies) [inline CHANGELOG.md]: I added this note and a link to the learn tutorial after screensharing with a number of users today who double clicked on the binary. 

1) Users shouldn't be double clicking on the binary to interact with the CLI, hence the learn link
2) Despite #1, it really IS confusing that a gatekeeper error appears when double clicking on the binary, whereas on other OS' the available commands are shown. We have a support ticket open with Apple about this, as this behavior isn't unique to terraform/hashi products- it happens with all go binaries on OSX 10.15+

Let me know if anything here should be clarified.

Comment (pselle) [reply CHANGELOG.md]: I'm wondering if we can simplify this to the link itself? (our recommended ...). Idk some thought about this being a long update and wanting to condense, considering we also have a fairly meaty update about destroy provisioners in this upcoming release.

Comment (pselle) [reply CHANGELOG.md]: Also needs a rebase now that those provisioner notes are in 🙈

Comment (mdeggies) [reply CHANGELOG.md]: Good call. I think it makes enough sense with just a line and the link. LMK how it looks now (merged in the master changes)

**Pull request merged**